okapi-core/src/main/java/org/folio/okapi/managers/TimerManager.java

```diff
@@ -0,0 +1,312 @@
+              fireTimer(tenantId, md, currentDescriptor);
+            }
+            // roll on.. wait and redo..
+            return waitTimer(tenantId, timerDescriptor);
+          });
+        })
+        .onFailure(cause -> logger.warn("handleTimer id={} {}", timerId,
+            cause.getMessage(), cause));
+  }
+
+  /**
+   * Handle a timer timer.
+   *
+   * <p>This method is called for each timer in each tenant and for each instance in
+   * the Okapi cluster. If the tenant descriptor has a zero delay, that will
+   * stop/disable the timer.
+   * @param tenantId tenant identifier
+   * @param timerDescriptor descriptor that this handling
+   */
+  private Future<Void> waitTimer(String tenantId, TimerDescriptor timerDescriptor) {
+    RoutingEntry routingEntry = timerDescriptor.getRoutingEntry();
+    final long delay = routingEntry.getDelayMilliSeconds();
+    logger.info("waitTimer {} delay {} for tenant {}", timerDescriptor.getId(), delay, tenantId);
+    if (delay > 0) {
+      vertx.setTimer(delay, res -> handleTimer(tenantId, timerDescriptor));
+    }
+    return Future.succeededFuture();
+  }
+
+  /**
+   * get timer descriptor.
+   * @param tenantID tenant identifier
+   * @param timerId timer identifier
+   * @return timer descriptor
+   */
+  public Future<TimerDescriptor> getTimer(String tenantID, String timerId) {
+    return tenantTimers.getNotFound(tenantID, timerId);
+  }
+
+  /**
+   * timer list.
+   * @param tenantId tenant identifier
+   * @return timer descriptors for the tenant
+   */
+  public Future<List<TimerDescriptor>> listTimers(String tenantId) {
+    return tenantTimers.get(tenantId).map(x -> x != null ? x : Collections.emptyList());
+  }
+
+  private static boolean isSimilar(TimerDescriptor a, TimerDescriptor b) {
+    if (a == null) {
+      return false;
+    }
+    return Json.encode(a).equals(Json.encode(b));
+  }
+
+  /**
+   * timer PATCH.
+   * @param tenantId tenant identifier
+   * @param timerDescriptor timer descriptor
+   * @return future
+   */
+  public Future<Void> patchTimer(String tenantId, TimerDescriptor timerDescriptor) {
+    return tenantTimers.getNotFound(tenantId, timerDescriptor.getId())
+        .compose(existing -> {
+          final String existingJson = Json.encode(existing);
+          final String timerId = timerDescriptor.getId();
+
+          RoutingEntry patchEntry = timerDescriptor.getRoutingEntry();
+          RoutingEntry existingEntry = existing.getRoutingEntry();
+
+          if (patchEntry.getDelay() == null && patchEntry.getUnit() == null
+              && patchEntry.getSchedule() == null) {
+            return tenantTimers.remove(tenantId, timerId).compose(deleted -> {
+              JsonObject o = new JsonObject();
+              o.put("tenantId", tenantId);
+              EventBus eb = vertx.eventBus();
+              eb.publish(EVENT_NAME, o.encode());
+              return Future.succeededFuture();
+            });
+          }
```

Comment:
Does this mean resetting one timer to default will trigger rescan all timers for a tenant?

Reply:
I have changed the strategy and simplified, so that reset and patch is same path.